File: Java/Transportation.java
abstract class Transportation
{
    private int vehicleId;
    private String vehicleName;
    private String engineType;

    Transportation(int vehicleId, String vehicleName, String engineType)
    {
        this.vehicleId = vehicleId;
        this.vehicleName = vehicleName;
        this.engineType = engineType;
    }

    public int getVehicleId()
    {
        return vehicleId;
    }

    public String getVehicleName()
    {
        return vehicleName;
    }

    public String getEngineType()
    {
        return engineType;
    }

    public void setVehicleId(int vehicleId)
    {
        this.vehicleId = vehicleId;
    }

    public void setVehicleName(String vehicleName)
    {
        this.vehicleName = vehicleName;
    }

    public void setEngineType(String engineType)
    {
        this.engineType = engineType;
    }

    public void displayDetails()
    {
        System.out.println("--------------- Vehicle Details ------------------");
        System.out.println("Vehicle ID     : " + vehicleId);
        System.out.println("Vehicle Name   : " + vehicleName);
        System.out.println("Engine Type    : " + engineType);
    }

    abstract void startJourney();
}

File: CS/Transportation.cs
using System;

public abstract class Transportation
{
    private int vehicleId;
    private string vehicleName;
    private string engineType;

    public Transportation(int id, string name, string engine)
    {
        vehicleId = id;
        vehicleName = name;
        engineType = engine;
    }

    public int GetVehicleId()
    {
        return vehicleId;
    }

    public string GetVehicleName()
    {
        return vehicleName;
    }

    public string GetEngineType()
    {
        return engineType;
    }

    public void SetVehicleId(int id)
    {
        vehicleId = id;
    }

    public void SetVehicleName(string name)
    {
        vehicleName = name;
    }

    public void SetEngineType(string engine)
    {
        engineType = engine;
    }

    public virtual void DisplayDetails()
    {
        Console.WriteLine("Vehicle ID     : " + vehicleId);
        Console.WriteLine("Vehicle Name   : " + vehicleName);
        Console.WriteLine("Engine Type    : " + engineType);
    }

    public abstract void StartJourney();
}